test/ShrinkingProps.hs

```diff
@@ -0,0 +1,519 @@
+{-# LANGUAGE BangPatterns        #-}
+{-# LANGUAGE DeriveAnyClass      #-}
+{-# LANGUAGE DeriveFoldable      #-}
+{-# LANGUAGE DeriveFunctor       #-}
+{-# LANGUAGE DeriveGeneric       #-}
+{-# LANGUAGE DeriveTraversable   #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE FlexibleInstances   #-}
+{-# LANGUAGE ImplicitParams      #-}
+{-# LANGUAGE PolyKinds           #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving  #-}
+{-# LANGUAGE TypeApplications    #-}
+{-# LANGUAGE TypeOperators       #-}
+
+module ShrinkingProps (
+    tests
+    -- Just to avoid compiler warnings
+  , ShrinkSeqFailure(..)
+  , ShrinkParFailure(..)
+  ) where
+
+import           Prelude                       hiding
+                   (elem)
+
+import           Control.Concurrent.STM
```

Comment:
Maybe use `import UnliftIO` or `import UnliftIO.STM` here, see #253. You can also drop the dependency on `stm`.